src/journal/src/Journal/Types.hs

```diff
@@ -130,6 +132,22 @@ tombStone = maxBound
 
 ------------------------------------------------------------------------
 
+newtype ReaderNotifier = ReaderNotifier { unReaderNotifier :: MVar ()}
```

Comment:
If we rename this to `WaitStrategy` and have another constructor `SpinLoopWaitMicros Int`, then `blockUntilNotification` can be renamed to `wait` and do `threadDelay` in the `SpinLoopWaitMacros` case, and `notifyReader` can be a noop in the spin loop case.